### PR TITLE
feat: storedby and last updated in enrollment analytics [DHIS2-11775]

### PR DESCRIPTION
**Pull request by gnespolino**
Refactored the time condition rendering a bit to accommodate enrollments.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        final SimpleDateFormat format = new SimpleDateFormat();
+        format.applyPattern( Period.DEFAULT_DATE_FORMAT );
+
+        String sql = "";
+        for ( String programStage : map.keySet() )
+        {
+            Set<AnalyticsPeriodBoundary> boundaries = map.get( programStage );
+
+            String eventTableName = "analytics_event_" + programIndicator.getProgram().getUid();
+            sql += " (select count(*) from " + eventTableName + " where " + eventTableName +
```
Comment (d-bernat) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java:148]: waht do you think about  replacement of  '... ( select count(*)..) > 0' with 'exists(select 1 from ... )'

**Review by d-bernat — CHANGES_REQUESTED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java (hunk 1 above)
Comment (gnespolino) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java:148]: makes sense and should improve performances. So generally speaking I agree with that.
However this code was just moved from somewhere else and is not meant to be modified within this ticket scope. 
I'd suggest to create a specific task for it

Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java:148]: Ok, agree

**Review by d-bernat — APPROVED** [review]

**Review by maikelarabori — APPROVED** [review]
Looks good in general :+1: I would just add some Javadocs on the main classes and methods.

**Pull request merged**